### ISSUE TITLE
Simple Model Training Utility

### ISSUE THREAD

**Issue by deprit** [twosixlabs/armory-library#153]
Armory Library users may need to train models as part of their workflows, for example, in evaluating the transferability of evasion attacks or the efficacy of poisoning defenses. The exemplar for this training utility will be the [Keras model training APIs](https://keras.io/api/models/model_training_apis/). 
- Model compile: package a PyTorch model with
    - a loss function,
    - an optimizer,
    - a learning rate scheduler,
    - any metrics,
    - callbacks such as early stopping.
- Model fit: fit the packaged model specifying
    - number of epochs
    - training and possibly validation data loader
Any metrics or artifacts produced by the model fit should be recorded as an experiment run in MLFlow.
